### PR TITLE
Add `pre-commit` check

### PR DESCRIPTION
**Pull request by stefanDeveloper**

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      id: pip-cache
+      with:
+        path: ${{ steps.get-pip-path.outputs.id }}
+        key: ${{ steps.get-pip-path.outputs.id }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements/requirements-dev.txt') }}-${{ hashFiles('requirements/requirements-logcollector.txt') }}-${{ hashFiles('requirements/requirements-detector.txt') }}-${{ hashFiles('requirements/requirements-prefilter.txt') }}-${{ hashFiles('requirements/requirements-inspector.txt') }}
+
+    - name: Install requirements
+      if: steps.pip-cache.outputs.cache-hit != 'true'
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements/requirements-dev.txt -r requirements/requirements.detector.txt -r requirements/requirements.logcollector.txt -r requirements/requirements.prefilter.txt -r requirements/requirements.inspector.txt
```
Comment (lamr02n) [inline .github/workflows/formatting.yml:39]: We have a new `requirements.logserver.txt`, which we need to add here too

File: .github/workflows/formatting.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      id: get-pip-path
+      run: |
+          id=$(echo ${{ env.pythonLocation }} | sed 's/\/bin//g')
+          echo "name=id::$id" >> $GITHUB_OUTPUT
+    - name: Pip cache
+      uses: actions/cache@v3
+      id: pip-cache
+      with:
+        path: ${{ steps.get-pip-path.outputs.id }}
+        key: ${{ steps.get-pip-path.outputs.id }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements/requirements-dev.txt') }}-${{ hashFiles('requirements/requirements-logcollector.txt') }}-${{ hashFiles('requirements/requirements-detector.txt') }}-${{ hashFiles('requirements/requirements-prefilter.txt') }}-${{ hashFiles('requirements/requirements-inspector.txt') }}
```
Comment (lamr02n) [inline .github/workflows/formatting.yml:33]: Same thing as line 39

File: docs/usage.rst [hunk 3]
```diff
@@ -27,9 +27,9 @@ Install all Python requirements.
...
 
 .. code-block:: console
-   
+
    $ source .venv/bin/activate
-   
+
 .. code-block:: console
 
    (.venv) $ pip install -r requirements/requirements-dev.txt -r requirements/requirements.detector.txt -r requirements/requirements.logcollector.txt -r requirements/requirements.prefilter.txt -r requirements/requirements.inspector.txt
```
Comment (lamr02n) [inline docs/usage.rst:35]: We're also missing the `requirements.logserver.txt` here.

**Review by lamr02n — CHANGES_REQUESTED** [review]
Looks good to me, we just need to add the LogServer's requirements file to some commands (see comments).

File: docs/usage.rst (hunk 3 above)
Comment (stefanDeveloper) [reply docs/usage.rst:35]: The `requirements-logserver.txt` has been added by #32 
I prefer to fix it in another PR.

File: .readthedocs.yml
Comment (lamr02n) [inline .readthedocs.yml:1]: I think the LogServer requirements are missing here

**Review by lamr02n — COMMENTED** [review]
Just one spot where the requirements are still missing

File: .readthedocs.yml
Comment (stefanDeveloper) [reply .readthedocs.yml:1]: This has been missing since #32. I will merge this PR now, since it is not related to my changes.

**Pull request merged**